nemo/collections/nlp/models/machine_translation/mt_enc_dec_model.py

```diff
@@ -456,71 +462,62 @@ def _setup_dataloader_from_config(self, cfg: DictConfig):
         if cfg.get("use_tarred_dataset", False):
             if cfg.get("metadata_file") is None:
                 raise FileNotFoundError("Trying to use tarred data set but could not find metadata path in config.")
-            else:
-                if not self.multilingual:
-                    metadata_file_list = [cfg.get('metadata_file')]
-                else:
-                    metadata_file_list = cfg.get('metadata_file')
-
-                datasets = []
-                for idx, metadata_file in enumerate(metadata_file_list):
-                    with open(metadata_file) as metadata_reader:
-                        metadata = json.load(metadata_reader)
-                    if cfg.get('tar_files') is None:
-                        tar_files = metadata.get('tar_files')
-                        if tar_files is not None:
-                            logging.info(f'Loading from tarred dataset {tar_files}')
-                        else:
-                            raise FileNotFoundError("Could not find tarred dataset in config or metadata.")
-                    else:
-                        tar_files = cfg.get('tar_files')
-                        if self.multilingual:
-                            tar_files = tar_files[idx]
-                        if metadata.get('tar_files') is not None:
-                            logging.info(
-                                f'Tar file paths found in both cfg and metadata using one in cfg by default - {tar_files}'
-                            )
-
-                    dataset = TarredTranslationDataset(
-                        text_tar_filepaths=tar_files,
-                        metadata_path=metadata_file,
-                        encoder_tokenizer=self.encoder_tokenizer,
-                        decoder_tokenizer=self.decoder_tokenizer,
-                        shuffle_n=cfg.get("tar_shuffle_n", 100),
-                        shard_strategy=cfg.get("shard_strategy", "scatter"),
-                        global_rank=self.global_rank,
-                        world_size=self.world_size,
-                        reverse_lang_direction=cfg.get("reverse_lang_direction", False),
-                        prepend_id=self.multilingual_ids[idx],
-                    )
-                    datasets.append(dataset)
-
-                if self.multilingual:
-                    dataset = ConcatDataset(
-                        datasets=datasets,
-                        sampling_technique=cfg.get('concat_sampling_technique'),
-                        sampling_temperature=cfg.get('concat_sampling_temperature'),
-                        sampling_probabilities=cfg.get('concat_sampling_probabilities'),
-                        global_rank=self.global_rank,
-                        world_size=self.world_size,
-                    )
+            metadata_file_list = cfg.get('metadata_file')
+            tar_files_list = cfg.get('tar_files', None)
+            if isinstance(metadata_file_list, str):
```

Comment:
Two options here:
1) Drop tar_files_list and tar_files params completely
2) If both ``metadata_file_list`` and ``tar_files_list`` raise an error and say only one needs to be specified.

Reply:
@aklife97 what do you think?

Reply:
I like option 1 here as I don't really find a good use for manually specifying the tar_files. Even if we have older datasets that don't have metadata_files, it still is probably better to generate metadata_files for them and do away with manually specifying every filename.

Reply:
I don't like 1. for the reason that tarfile paths in the metadata file are only valid on the place where the tarred dataset was created. I want to be able to move tarfiles around from my workstation to different clusters and then point to paths there.

Reply:
then let's just do 2.

Reply:
Even in 2, both need to be specified because the metadata file contains information about num_batches and tar_files_list contains paths to the tarfiles it isn't possible to specify just one unless the metadata file contains valid paths on the env in which it is running.

Reply:
@okuchaiev, @aklife97, and @ericharper hope it is fine with all of you to keep things as they are now to allow pre-processing on a certain machine and training on another.